TESTS/netsocket/udp/udpsocket_echotest.cpp

```diff
@@ -183,20 +147,37 @@ void UDPSOCKET_ECHOTEST_NONBLOCK()
                 printf("[Round#%02d - Sender] error, returned %d\n", s_idx, sent);
                 continue;
             }
-            if (!tx_sem.try_acquire_for(WAIT2RECV_TIMEOUT * 2)) { // RX might wait up to WAIT2RECV_TIMEOUT before recvfrom
-                continue;
+
+            int recvd;
+            for (int retry_recv = 0; retry_recv <= RETRIES; retry_recv++) {
+                recvd = sock->recvfrom(NULL, rx_buffer, pkt_s);
+                if (recvd == NSAPI_ERROR_WOULD_BLOCK) {
+                    if (tc_exec_time.read() >= time_allotted) {
+                        break;
+                    }
+                    signals.wait_all(SIGNAL_SIGIO_RX, SIGIO_TIMEOUT);
```

Comment:
The previous loop uses `osSignalWait(SIGNAL_SIGIO_TX, SIGIO_TIMEOUT).status == osEventTimeout)` which will wait that signal send to current thread.
However, I don't see anyone sending that signal.

The `_sigio_handler()` seems to set flags only on this `EventFlags signals` variable.

Please fix the previous loop to use the same eventflag variable.

Reply:
Changed